### PR TITLE
CONDEC-775: Integrate decision table view in decision knowledge page

### PR DESCRIPTION
**Pull request by kleebaum**

### CODE REVIEW

File: src/main/java/de/uhd/ifi/se/decision/management/jira/view/decisiontable/DecisionTable.java [hunk 1]
```diff
@@ -89,16 +89,19 @@ public void setIssues(FilterSettings filterSettings, ApplicationUser user) {
...
-			KnowledgeElement targetElement = currentLink.getTarget();
-			if (targetElement.getType() == KnowledgeType.ALTERNATIVE
-					|| targetElement.getType() == KnowledgeType.DECISION) {
-				decisionTableData.get("alternatives").add(new Alternative(targetElement));
-				getArguments(targetElement);
+		// TODO Check link type. A decision that leads to a new decision problem should
+		// not be shown as solution option for this derived decision problem.
+		for (Link currentLink : graph.edgesOf(rootElement)) {
+			KnowledgeElement oppositeElement = currentLink.getOppositeElement(rootElement);
+			System.out.println(oppositeElement);
```
Comment (zubrod) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/view/decisiontable/DecisionTable.java:99]: Should there be printed lines in the Java Code?

File: src/main/java/de/uhd/ifi/se/decision/management/jira/view/decisiontable/DecisionTable.java [hunk 2]
```diff
@@ -132,12 +135,14 @@ public void getArgumentCriteria(Argument argument, List<DecisionTableElement> cr
 		// persistenceManager
 		KnowledgeElement rootElement = persistenceManager.getKnowledgeElement(argument.getId(),
 				argument.getDocumentationLocation());
-		Set<Link> outgoingLinks = this.graph.outgoingEdgesOf(rootElement);
+		Set<Link> linksOfArgument = this.graph.edgesOf(rootElement);
 
-		for (Link currentLink : outgoingLinks) {
-			KnowledgeElement elem = currentLink.getTarget();
+		for (Link currentLink : linksOfArgument) {
+			KnowledgeElement elem = currentLink.getOppositeElement(rootElement);
```
Comment (zubrod) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/view/decisiontable/DecisionTable.java:141]: getOpposite() could be null 
image

File: src/main/resources/js/condec.decision.table.js [hunk 3]
```diff
@@ -3,34 +3,39 @@
...
+		this.updateView();
 	};
 	
-	function addOnClickEventToDecisionTableButtons () {
-		document.getElementById("btnAddCriterion").addEventListener("click", function (event) {
-			conDecDecisionTable.showAddCriteriaToDecisionTableDialog();
-		});
-		document.getElementById("btnAddAlternative").addEventListener("click", function (event) {
+	ConDecDecisionTable.prototype.updateView = function () {
+		this.loadDecisionProblems();
```
Comment (zubrod) [inline src/main/resources/js/condec.decision.table.js:27]: image

You could just reload the table data and without reloading the dropdownmenu

Comment (kleebaum) [reply src/main/resources/js/condec.decision.table.js:25]: I implemented updateView like this:
`ConDecDecisionTable.prototype.updateView = function () {
		this.loadDecisionProblems();
	};`

File: src/main/java/de/uhd/ifi/se/decision/management/jira/view/decisiontable/DecisionTable.java (hunk 1 above)
Comment (kleebaum) [reply src/main/java/de/uhd/ifi/se/decision/management/jira/view/decisiontable/DecisionTable.java:99]: No, there should be no printed lines in the Java Code, absolutely not!

**Pull request merged**